File: src/main/java/io/primeaspect/calculator/repository/mapper/ExpressionMapper.java
package io.primeaspect.calculator.repository.mapper;

import io.primeaspect.calculator.model.Expression;
import org.apache.ibatis.annotations.*;

import java.time.LocalDate;
import java.util.List;

@Mapper
public interface ExpressionMapper {
    @Insert("INSERT INTO expression (date, expression, solution) VALUES " +
            "(#{expression.date}, #{expression.expression}, #{expression.solution})")
    void saveExpression(@Param("expression") Expression expression);

    @Select("SELECT * FROM expression")
    List<Expression> findAll();

    @Select("SELECT * FROM expression WHERE date=#{date}")
    List<Expression> findAllByDate(LocalDate date);

    @Select("SELECT * FROM expression WHERE expression LIKE '%${operation}%'")
    List<Expression> findAllByOperation(String operation);

    @Select("SELECT COUNT(*) FROM expression WHERE date=#{date}")
    int getCountByDate(LocalDate date);

    @Select("SELECT COUNT(*) FROM expression WHERE expression LIKE '%${operation}%'")
    int getCountByOperation(String operation);

    @Delete("DELETE FROM expression")
    void deleteAllExpressions();
}

File: src/main/kotlin/io/primeaspect/calculator/repository/impl/mybatis/mapper/ExpressionMapper.kt
package io.primeaspect.calculator.repository.impl.mybatis.mapper

import io.primeaspect.calculator.model.Expression
import org.apache.ibatis.annotations.*
import java.time.LocalDate

@Mapper
interface ExpressionMapper {
    @Insert(
            "INSERT INTO expression (date, expression, solution) VALUES " +
                    "(#{expression.date}, #{expression.expression}, #{expression.solution})"
    )
    fun saveExpression(@Param("expression") expression: Expression?)

    @Select("SELECT * FROM expression")
    fun findAll(): MutableList<Expression>

    @Select("SELECT * FROM expression WHERE date=#{date}")
    fun findAllByDate(date: LocalDate?): MutableList<Expression>

    @Select("SELECT * FROM expression WHERE expression LIKE '%\${operation}%'")
    fun findAllByOperation(operation: String?): MutableList<Expression>

    @Select("SELECT COUNT(*) FROM expression WHERE date=#{date}")
    fun getCountByDate(date: LocalDate?): Int

    @Select("SELECT COUNT(*) FROM expression WHERE expression LIKE '%\${operation}%'")
    fun getCountByOperation(operation: String?): Int

    @Delete("DELETE FROM expression")
    fun deleteAllExpressions()
}